docs/workflows/accessing-payload-data.md

```diff
@@ -0,0 +1,330 @@
+*   `sqrt` (computes the square root of a number)
+*   `log` (computes the natural logarithm of a number)
+*   `log10` (computes the base 10 logarithm of a number)
+*   `abs` (computes the absolute value of a number)
+*   `ceil` (computes the smallest integer greater than or equal to a number)
+*   `floor` (computes the largest integer less than or equal to a number)
+*   `round` (rounds a number to the nearest integer)
+*   `trunc` (truncates any fractional digits of a number)
+*   `exp` (computes e raised to the power of the given number)
+*   `pow` (raises a number to the power of a second number)
+*   `length` (computes the length of an array, object, or string)
+
+### Examples
+
+Given our example object, let's see how these expressions evaluate.
+
+```javascript
+{{data.aNumberProp}} // true
+{{data.anUndefinedProp}} // false
+{{data.aBooleanProp}} // true
+!{{data.aBooleanProp}} // false
+
+{{data.aNumberProp}} === 'hello' // false
+{{data.aNumberProp}} > 20 // true
+
+{{data}} && {{data.aBooleanProp}} // true
+{{data.aBooleanProp}} || {{data.aFalsyProp}} // true
+{{data.aBooleanProp}} && {{data.aFalsyProp}} // false
+
+{{data.aNumberProp}} > {{data.anObject.aNestedArray.[1]}} // true
+{{data.aNumberProp}} > {{data.aStringProp}} // false
+{{data.aNumberProp}} < {{data.aStringProp}} // false
+
+{{data.aFalsyProp}} == false // true
+{{data.aFalsyProp}} === false // false
+
+{{format data.aJavascriptDateObject 'MMMM'}} === 'September' // true
+{{lower data.aStringProp}} === 'Hello world' // false
+
+{{data.anObject.aNestedArray.[1]}} / 5 <= PI // false
+pow({{data.aNumberProp}}, {{data.aFalsyProp}}) === 1 // true
+```
+
+## JSON Templates
+
+In a few places within the Losant platform – such as the [Input Controls block](/dashboards/input-controls/) and [Losant API node](/workflows/data/losant-api/), it is necessary to construct JSON objects from other data. In these instances we use **JSON templates**.
+
+JSON templates can take any of the following formats (or a combination thereof):
+
+*   Valid, static JSON. e.g. `{"foo": "bar"}`, `[1,2,3]` or `null`
+*   JSON containing block helpers around keys, values or both. e.g. `{ {{#if foo}}"bar": "bat"{{else}}"baz": "bub"{{/if}} }`
+*   JSON containing string helpers for keys or values. e.g. `{"month": {{format date 'MMMM'}} }`
+*   A single reference to a property on an object whose value is an object. e.g. `{{foo.bar}}`
+
+Your entire input will run through Handlebars and the Losant-provided helpers. **After evaluation, the result must be valid JSON.** Because of this, any string value being output by a string template should be wrapped in double quotes, like so: `{ "foo": "{{data.aStringProp}}" }`.
+
+### Examples
+
+Again, given our example object above, these JSON templates will evaluate as follows ...
+
+```javascript
+{ "foo": "bar", "staticNumber" : 66 }
+// { "foo": "bar", "staticNumber" : 66 }
+
+{ "myArray": {{data.anArray}} }
```

Comment:
if you think this *should* work this way, I can probably be convinced - we can have a discussion about it.  would essentially json encode arrays/objects by default?